internal/k8s/validation_test.go

```diff
@@ -150,19 +161,185 @@ func TestValidateIngressAnnotations(t *testing.T) {
+			},
+			msg: "invalid nginx.com/health-checks-mandatory, must be a boolean",
+		},
+		{
+			annotations: map[string]string{
+				"nginx.com/health-checks-mandatory": "true",
+			},
+			isPlus: true,
+			expectedErrors: []string{
+				"annotations.nginx.com/health-checks-mandatory: Forbidden: related annotation nginx.com/health-checks: must be set",
+			},
+			msg: "invalid nginx.com/health-checks-mandatory, related annotation nginx.com/health-checks not set",
+		},
+		{
+			annotations: map[string]string{
+				"nginx.com/health-checks":           "false",
+				"nginx.com/health-checks-mandatory": "true",
+			},
+			isPlus: true,
+			expectedErrors: []string{
+				"annotations.nginx.com/health-checks-mandatory: Forbidden: related annotation nginx.com/health-checks: must be true",
+			},
+			msg: "invalid nginx.com/health-checks-mandatory nginx.com/health-checks is not true",
+		},
+
+		{
+			annotations: map[string]string{
+				"nginx.com/health-checks":                 "true",
+				"nginx.com/health-checks-mandatory":       "true",
+				"nginx.com/health-checks-mandatory-queue": "5",
+			},
+			isPlus:         true,
+			expectedErrors: nil,
+			msg:            "valid nginx.com/health-checks-mandatory-queue annotation",
+		},
+		{
+			annotations: map[string]string{
+				"nginx.com/health-checks":                 "true",
+				"nginx.com/health-checks-mandatory":       "true",
+				"nginx.com/health-checks-mandatory-queue": "not_a_number",
+			},
+			isPlus: true,
+			expectedErrors: []string{
+				"annotations.nginx.com/health-checks-mandatory-queue: Invalid value: \"not_a_number\": must be a non-negative integer",
+			},
+			msg: "invalid nginx.com/health-checks-mandatory-queue, must be a boolean",
+		},
+		{
+			annotations: map[string]string{
+				"nginx.com/health-checks-mandatory-queue": "5",
+			},
+			isPlus: true,
+			expectedErrors: []string{
+				"annotations.nginx.com/health-checks-mandatory-queue: Forbidden: related annotation nginx.com/health-checks-mandatory: must be set",
+			},
+			msg: "invalid nginx.com/health-checks-mandatory-queue, related annotation nginx.com/health-checks-mandatory not set",
+		},
+		{
+			annotations: map[string]string{
+				"nginx.com/health-checks":                 "true",
+				"nginx.com/health-checks-mandatory":       "false",
+				"nginx.com/health-checks-mandatory-queue": "5",
+			},
+			isPlus: true,
+			expectedErrors: []string{
+				"annotations.nginx.com/health-checks-mandatory-queue: Forbidden: related annotation nginx.com/health-checks-mandatory: must be true",
+			},
+			msg: "invalid nginx.com/health-checks-mandatory-queue nginx.com/health-checks-mandatory is not true",
+		},
 	}
 
 	for _, test := range tests {
-		allErrs := validateIngressAnnotations(test.annotations, field.NewPath("annotations"))
-		assertion := assertErrors("validateIngressAnnotations()", test.msg, allErrs, test.expectedErrors)
-		if assertion != "" {
-			t.Error(assertion)
-		}
+		t.Run(test.msg, func(t *testing.T) {
```

Comment:
any reason for running each test as a subset? asking that, because the other similar tests in `validation_test.go` don't use that approach.

Reply:
Is there a downside to doing this? The reason I did it is that Goland will report individual table test failures with this change, and let you run individual table test cases. IDE support is usually a terrible reason to change the code, so let me know if you think this should be removed, no problem.

Reply:
Seems there are some nice benefits. See the examples: https://blog.golang.org/subtests#TOC_5.

However if we do this change, it might make sense to do the change consistently in a separate PR.

Reply:
Yeah I think a separate PR would be better if the team feel it is a worthwhile change to make.